Tools/cases_generator/generate_cases.py

```diff
@@ -156,16 +156,9 @@ def __init__(
         self.emit_line_directives = emit_line_directives
         self.comment = comment
         self.lineno = 1
-        filename = os.path.relpath(self.stream.name, ROOT)
-        # Make filename more user-friendly and less platform-specific
-        filename = filename.replace("\\", "/")
-        if filename.startswith("./"):
-            filename = filename[2:]
-        if filename.endswith(".new"):
-            filename = filename[:-4]
-        self.filename = filename
+        self.filename = prettify_filename(self.stream.name)
```

Comment:
These are not really improvements, but this is my attempt to fix tests on Windows.
As I said, I am not a Win user, so my approach might be incorrect.

But, the root cause for this is that https://github.com/python/cpython/actions/runs/5542920427/jobs/10118249499 fails with:

```
Traceback (most recent call last):
  File "D:\a\cpython\cpython\Lib\test\test_generated_cases.py", line 327, in test_suppress_dispatch
    self.run_cases_test(input, output)
  File "D:\a\cpython\cpython\Lib\test\test_generated_cases.py", line 109, in run_cases_test
    a.write_instructions()
  File "D:\a\cpython\cpython\Tools\cases_generator\generate_cases.py", line 1448, in write_instructions
    self.out = Formatter(f, 8, self.emit_line_directives)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\a\cpython\cpython\Tools\cases_generator\generate_cases.py", line 159, in __init__
    filename = os.path.relpath(self.stream.name, ROOT)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\a\cpython\cpython\Lib\ntpath.py", line 779, in relpath
    raise ValueError("path is on mount %r, start on mount %r" % (
ValueError: path is on mount 'C:', start on mount 'D:'
```

This happens because temp dir is on another drive and Windows cannot create a rel path to it. If there are other solutions, I am happy to apply them instead.

Reply:
Okay, that's fair. Let's keep this version then.